src/iceberg/iceberg_parser/iceberg_parser.py

```diff
@@ -0,0 +1,208 @@
+"""
+Author: Ioannis Paraskevakos
+License: MIT
+Copyright: 2018-2019
+"""
+
+from __future__ import print_function
+import argparse
+import sys
+
+
+class IcebergParser(object):
+    """
+    This class is the argument parser of the ICEBERG software tool.
+    """
+
+    def __init__(self):
+        """
+        The constructor
+        """
+        parser = argparse.ArgumentParser(description='ICEBERG command description:',
+                                         epilog='''Enjoy our tool!''')
+        required_args = parser.add_argument_group()
+        required_args.title = 'Required Arguments'
+        required_args.add_argument('--resource', '-r', help='Where the execution will happen',
+                                   type=str)
+        required_args.add_argument('--queue', '-q', help='The queue of the resource',
+                                   type=str)
+        required_args.add_argument('--cpus', '-c', help='How many CPUs will be required',
+                                   type=int)
+        required_args.add_argument('--gpus', '-g', help='How many GPUs will be required',
+                                   type=int)
+        required_args.add_argument('--input_path', '-ip', help='Where the input images are',
+                                   type=str)
+        required_args.add_argument('--output_path', '-op', help='Where the results should be saved',
+                                   type=str)
+        required_args.add_argument('--walltime', '-w', help='The estimated execution time',
+                                   type=int)
+        required_args.add_argument('--analysis', '-a', help='The type of analysis to be executed',
+                                   type=str)
+
+        if len(sys.argv) < 9:
+            parser.print_help()
+            if '--help' in sys.argv or '-h' in sys.argv:
+                argv = sys.argv.remove('--help')
+            elif '-h' in sys.argv:
+                argv = sys.argv.remove('-h')
+            args = parser.parse_args(argv)
```

Comment:
I am doing this ```iceberg``` and getting this error at the end of the ```ICEBERG command description: ```

```
Traceback (most recent call last):
  File "/home/aymen/miniconda3/envs/conda_env1/bin/iceberg", line 12, in <module>
    IcebergParser()
  File "/home/aymen/miniconda3/envs/conda_env1/lib/python2.7/site-packages/iceberg/iceberg_parser/iceberg_parser.py", line 48, in __init__
    args = parser.parse_args(argv)
UnboundLocalError: local variable 'argv' referenced before assignment

```

Reply:
That's because of a logical bug! Thank you @AymenFJA for catching it!

Reply:
so, I did the following :

1. git pull.
2. pip install .  -upgrade 
3. ```iceberg```

and i am getting the same error above .